File: Tests/PrimitivePage.m
//
// microtome - Copyright 2012 Three Rings Design

#import "PrimitivePage.h"

static NSString* const XML_STRING =
    @"<primitiveTest type='PrimitivePage'>"
    @"  <foo>true</foo>"
    @"  <bar>2</bar>"
    @"  <baz>3.1415</baz>"
    @"</primitiveTest>";

@implementation PrimitivePage {
@protected
    MTMutableBoolProp* _foo;
    MTMutableIntProp* _bar;
    MTMutableFloatProp* _baz;
}

+ (NSString*)XML { return XML_STRING; }

- (BOOL)foo { return _foo.value; }
- (int)bar { return _bar.value; }
- (float)baz { return _baz.value; }

- (id)init {
    if ((self = [super init])) {
        _foo = [[MTMutableBoolProp alloc] initWithName:@"foo" parent:self];
        _bar = [[MTMutableIntProp alloc] initWithName:@"bar" parent:self];
        _baz = [[MTMutableFloatProp alloc] initWithName:@"baz" parent:self];
    }
    return self;
}

- (NSArray*)props {
    return MT_PROPS(_foo, _bar, _baz);
}

@end

File: runtime/src/test/py/primitive_page.py
#
# microtome

from microtome.page import Page
from microtome.prop.prop_spec import PropSpec
from microtome.prop.prop import Prop

class PrimitivePage(Page):
    def __init__(self, name):
        super(PrimitivePage, self).__init__(name)
        self._foo = Prop(self, PrimitivePage._s_fooSpec)
        self._bar = Prop(self, PrimitivePage._s_barSpec)
        self._baz = Prop(self, PrimitivePage._s_bazSpec)

    @property
    def foo(self):
        return self._foo.value

    @property
    def bar(self):
        return self._bar.value

    @property
    def baz(self):
        return self._baz.value

    @property
    def props(self):
        return super(PrimitivePage, self).props + [self._foo, self._bar, self._baz]

    _s_fooSpec = PropSpec("foo", None, [bool, ])
    _s_barSpec = PropSpec("bar", None, [int, ])
    _s_bazSpec = PropSpec("baz", None, [float, ])
